src/base/logger.cpp

```diff
@@ -82,13 +82,25 @@ void Logger::addMessage(const QString &message, const Log::MsgType &type)
 void Logger::addPeer(const QString &ip, const bool blocked, const QString &reason)
 {
     QWriteLocker locker(&m_lock);
-    const Log::Peer msg = {m_peerCounter++, QDateTime::currentMSecsSinceEpoch(), ip.toHtmlEscaped(), blocked, reason.toHtmlEscaped()};
+    const Log::Peer msg = {m_peerCounter++, QDateTime::currentMSecsSinceEpoch(), ip, blocked, reason};
     m_peers.push_back(msg);
     locker.unlock();
 
     emit newLogPeer(msg);
 }
 
+Log::Msg Logger::message(const int index) const
+{
+    QReadLocker locker(&m_lock);
```

Comment:
`Logger::message()` and `Logger::peer()` read data from different source so they should have different lock.

Reply:
done